docs/content/about-iota/iota-architecture/protocol-upgrades.mdx

```diff
@@ -6,54 +6,46 @@ import {questions} from '../../../site/static/json/about-iota/iota-architecture/
 
 # Protocol Upgrades
 
-The IOTA protocol, framework, and execution engine are frequently updated to include new features and bug fixes. This
-new functionality is released as part of our regular software updates to validator operators. However, the IOTA protocol
-requires all validators to agree on the results of executing each transaction.
+The IOTA protocol, framework, and execution engine are frequently updated to introduce new features and bug fixes. These updates are released as part of regular software updates for validator operators. However, since all validators must agree on the results of each transaction execution, introducing changes to transaction execution presents a challenge.
 
-This creates a challenge: How to release code that changes transaction execution, given that it's impossible to
-ensure all operators upgrade their software simultaneously? 
-Additionally, how to ensure that all IOTA transaction
-history can be replayed even after functionality has changed?
+- How to ensure that all validators upgrade their software in a coordinated manner?
 
-To address this, IOTA uses a process called protocol upgrades.
+- How to ensure that the transaction history remains consistent and replayable despite protocol changes?
+
+IOTA addresses these challenges through a structured **protocol upgrade** process.
 
 ## Protocol Upgrade Process
 
-The protocol upgrade process includes the following steps:
+### Protocol Versioning
+
+Validators run the `iota-node` software, which supports multiple protocol versions, including future versions. The protocol version is determined by the `ProtocolConfig` struct. Validators that do not support the current protocol version cannot participate in consensus.
+
+The protocol version to be executed in the next epoch is recorded on-chain in a special end-of-epoch transaction. This ensures that all validators switch to the new protocol version at the same time.
+
+### Upgrade Steps
+
+1. **Implementation**: IOTA developers introduce a new protocol version by updating the `ProtocolConfig` struct, which may include feature flags to control specific functionalities within the new version.
```

Comment:
Maybe we could also make it a little bit more complete by adding something like:

```suggestion
1. **Implementation**: IOTA developers introduce a new protocol version by updating the `ProtocolConfig` struct with properties and values, which may include feature flags to control specific functionalities within the new version.
```